gobblin-service/src/test/java/gobblin/service/modules/core/GobblinServiceHATest.java

```diff
@@ -0,0 +1,501 @@
+      Assert.fail("Get should have gotten a 404 error");
+    } catch (RestLiResponseException e) {
+      Assert.assertEquals(e.getStatus(), HttpStatus.NOT_FOUND_404);
+    }
+
+    try {
+      this.node2FlowConfigClient.getFlowConfig(flowConfigId);
+      Assert.fail("Get should have gotten a 404 error");
+    } catch (RestLiResponseException e) {
+      Assert.assertEquals(e.getStatus(), HttpStatus.NOT_FOUND_404);
+    }
+  }
+
+  @Test (dependsOnMethods = "testDelete")
+  public void testBadGet() throws Exception {
+    FlowConfigId flowConfigId = new FlowConfigId().setFlowGroup(TEST_DUMMY_GROUP_NAME_1).setFlowName(TEST_DUMMY_FLOW_NAME_1);
+
+    try {
+      this.node1FlowConfigClient.getFlowConfig(flowConfigId);
+      Assert.fail("Get should have raised a 404 error");
+    } catch (RestLiResponseException e) {
+      Assert.assertEquals(e.getStatus(), HttpStatus.NOT_FOUND_404);
+    }
+
+    try {
+      this.node2FlowConfigClient.getFlowConfig(flowConfigId);
+      Assert.fail("Get should have raised a 404 error");
+    } catch (RestLiResponseException e) {
+      Assert.assertEquals(e.getStatus(), HttpStatus.NOT_FOUND_404);
+    }
+  }
+
+  @Test (dependsOnMethods = "testBadGet")
+  public void testBadDelete() throws Exception {
+    FlowConfigId flowConfigId = new FlowConfigId().setFlowGroup(TEST_DUMMY_GROUP_NAME_1).setFlowName(TEST_DUMMY_FLOW_NAME_1);
+
+    try {
+      this.node1FlowConfigClient.getFlowConfig(flowConfigId);
+      Assert.fail("Get should have raised a 404 error");
+    } catch (RestLiResponseException e) {
+      Assert.assertEquals(e.getStatus(), HttpStatus.NOT_FOUND_404);
+    }
+
+    try {
+      this.node2FlowConfigClient.getFlowConfig(flowConfigId);
+      Assert.fail("Get should have raised a 404 error");
+    } catch (RestLiResponseException e) {
+      Assert.assertEquals(e.getStatus(), HttpStatus.NOT_FOUND_404);
+    }
+  }
+
+  @Test (dependsOnMethods = "testBadDelete")
+  public void testBadUpdate() throws Exception {
+    Map<String, String> flowProperties = Maps.newHashMap();
+    flowProperties.put("param1", "value1b");
+    flowProperties.put("param2", "value2b");
+
+    FlowConfig flowConfig = new FlowConfig().setFlowGroup(TEST_DUMMY_GROUP_NAME_1).setFlowName(TEST_DUMMY_FLOW_NAME_1)
+        .setTemplateUris(TEST_TEMPLATE_URI_1).setSchedule(TEST_SCHEDULE_1).setProperties(new StringMap(flowProperties));
+
+    try {
+      this.node1FlowConfigClient.updateFlowConfig(flowConfig);
+      Assert.fail("Get should have raised a 404 error");
+    } catch (RestLiResponseException e) {
+      Assert.assertEquals(e.getStatus(), HttpStatus.NOT_FOUND_404);
+    }
+
+    try {
+      this.node2FlowConfigClient.updateFlowConfig(flowConfig);
+      Assert.fail("Get should have raised a 404 error");
+    } catch (RestLiResponseException e) {
+      Assert.assertEquals(e.getStatus(), HttpStatus.NOT_FOUND_404);
+    }
+  }
+
+  @Test (dependsOnMethods = "testBadUpdate")
```

Comment:
Travis is slow, so that may affect the failover time. First see if it can run reliably on travis in a reasonable amount of time.